File: app/src/main/java/com/mxt/anitrend/model/entity/base/NotificationBase.java
package com.mxt.anitrend.model.entity.base;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.Nullable;

import com.mxt.anitrend.model.entity.group.RecyclerItem;
import com.mxt.anitrend.util.KeyUtil;

/**
 * Created by max on 2018/02/24.
 * Notification base meta data class
 */
public class NotificationBase extends RecyclerItem implements Parcelable {

    private long id;
    private @KeyUtil.NotificationType String type;
    private long createdAt;
    private String context;

    protected NotificationBase(Parcel in) {
        id = in.readLong();
        type = in.readString();
        createdAt = in.readLong();
        context = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeLong(id);
        dest.writeString(type);
        dest.writeLong(createdAt);
        dest.writeString(context);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<NotificationBase> CREATOR = new Creator<NotificationBase>() {
        @Override
        public NotificationBase createFromParcel(Parcel in) {
            return new NotificationBase(in);
        }

        @Override
        public NotificationBase[] newArray(int size) {
            return new NotificationBase[size];
        }
    };

    public long getId() {
        return id;
    }

    @Nullable
    public @KeyUtil.NotificationType String getType() {
        return type;
    }

    public long getCreatedAt() {
        return createdAt;
    }

    public String getContext() {
        return context.trim();
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof NotificationBase)
            return ((NotificationBase)obj).getId() == id;
        return super.equals(obj);
    }
}

File: app/src/main/java/com/mxt/anitrend/model/entity/base/NotificationBase.kt
package com.mxt.anitrend.model.entity.base

import android.os.Parcel
import android.os.Parcelable
import com.mxt.anitrend.model.entity.group.RecyclerItem
import com.mxt.anitrend.util.KeyUtil

/**
 * Created by max on 2018/02/24.
 * Notification base meta data class
 */
open class NotificationBase() :
    RecyclerItem(),
    Parcelable {
    var id: Long = 0

    @KeyUtil.NotificationType
    var type: String? = null

    var createdAt: Long = 0

    var context: String = ""
        get() = field.trim()

    protected constructor(parcel: Parcel) : this() {
        id = parcel.readLong()
        type = parcel.readString()
        createdAt = parcel.readLong()
        context = parcel.readString().orEmpty()
    }

    override fun writeToParcel(
        dest: Parcel,
        flags: Int,
    ) {
        dest.writeLong(id)
        dest.writeString(type)
        dest.writeLong(createdAt)
        dest.writeString(context)
    }

    override fun describeContents(): Int = 0

    override fun equals(other: Any?): Boolean {
        if (other is NotificationBase) {
            return other.id == id
        }
        return super.equals(other)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<NotificationBase> =
            object : Parcelable.Creator<NotificationBase> {
                override fun createFromParcel(parcel: Parcel): NotificationBase = NotificationBase(parcel)

                override fun newArray(size: Int): Array<NotificationBase?> = arrayOfNulls(size)
            }
    }
}
